src/distributions/binomial.rs

```diff
@@ -90,20 +89,14 @@ impl Distribution<u64> for Binomial {
 
         let mut lresult;
 
+        // we use the Cauchy distribution as the comparison distribution
+        // f(x) ~ 1/(1+x^2)
+        let cauchy = Cauchy::new(0.0, 1.0);
         loop {
-            let mut comp_dev: f64;
-            // we use the lorentzian distribution as the comparison distribution
-            // f(x) ~ 1/(1+x/^2)
-            loop {
-                // draw from the lorentzian distribution
-                comp_dev = (PI*rng.gen::<f64>()).tan();
-                // shift the peak of the comparison ditribution
-                lresult = expected + sq * comp_dev;
-                // repeat the drawing until we are in the range of possible values
-                if lresult >= 0.0 && lresult < float_n + 1.0 {
-                    break;
-                }
```

Comment:
Surely this clamp on the output is there for a reason and removing it doesn't make sense? It traps for the π/2 value but also for negatives and large results. (I don't know what is needed here, but do know this is not the same code.)

Reply:
Many of the tests aren't exhaustive though. I don't really understand how this code works, and if you don't either I think we shouldn't adjust what it does. I think it's still possible to use the Cauchy code here but not sure whether it's worth it.

Reply:
From some benchmarking I just did it looks like using Cauchy (with the loop put back in) is ~9000 nanoseconds slower than the existing code because I check if the rng produced 0.5 before using it, whereas the existing code does not. If I remove the check it has similar performance.

Since the Cauchy distribution is getting used in more than one place in the codebase I think there is a benefit to standardizing the generation of it.